File: shared/contracts/learning_agility.ts
export type LearningAgilityTransition = {
  from_session_id: string
  to_session_id: string
  completed_at: string
  next_request_created_at: string
  days_to_apply: number
  agility_score: number
  conceptual_avg: number | null
  technical_avg: number | null
  originality_avg: number | null
}

export type LearningAgilityDisciplineBreakdown = {
  discipline: string
  transitions: LearningAgilityTransition[]
  avg_days_to_apply: number | null
  avg_agility_score: number | null
}

export type LearningAgilityMineResponse = {
  target_days: number
  disciplines: LearningAgilityDisciplineBreakdown[]
  overall_avg_days_to_apply: number | null
  overall_avg_agility_score: number | null
  transitions_count: number
}

File: backend/app/schemas/learning_agility.py
import datetime as dt
from typing import Literal, Optional

from pydantic import BaseModel, Field


class LearningAgilityTransition(BaseModel):
    from_session_id: str
    to_session_id: str

    completed_at: dt.datetime
    next_request_created_at: dt.datetime

    days_to_apply: float = Field(gt=0)
    agility_score: float = Field(ge=0)

    conceptual_avg: Optional[int] = None
    technical_avg: Optional[int] = None
    originality_avg: Optional[int] = None


class LearningAgilityDisciplineBreakdown(BaseModel):
    discipline: str
    transitions: list[LearningAgilityTransition]

    avg_days_to_apply: Optional[float] = None
    avg_agility_score: Optional[float] = None


class LearningAgilityMineResponse(BaseModel):
    target_days: float
    disciplines: list[LearningAgilityDisciplineBreakdown]

    overall_avg_days_to_apply: Optional[float] = None
    overall_avg_agility_score: Optional[float] = None
    transitions_count: int


LearningAgilityStatus = Literal["queued", "assigned", "completed"]

